bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/sitequality/TimingWindowDialog.java

```diff
@@ -0,0 +1,285 @@
+package jsky.app.ot.gemini.editor.sitequality;
+
+import edu.gemini.shared.util.HourMinuteFormat;
+import edu.gemini.shared.util.immutable.*;
+import edu.gemini.spModel.gemini.obscomp.SPSiteQuality.TimingWindow;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.text.*;
+import java.util.Date;
+import java.util.TimeZone;
+
+class TimingWindowDialog extends JDialog {
+    private static final long MS_PER_MINUTE = 1000 * 60;
+    private static final long MS_PER_HOUR   = MS_PER_MINUTE * 60;
+
+    private static final String INSTRUCTIONS =
+        "Specify a timing window. Starting time should be in the form YYYY-MM-DD hh:mm:ss " +
+        "and will be interpreted in UTC. Window duration and repeat period are specified in " +
+        "hours and minutes.";
+
+    private static final SimpleDateFormat UTC   = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss") {{
+        setTimeZone(TimeZone.getTimeZone("UTC"));
+    }};
+    private static final NumberFormat     TIMES = new DecimalFormat("######");
+
+
+    private final JFormattedTextField window;
+    private final JFormattedTextField duration;
+    private final JFormattedTextField period;
+    private final JFormattedTextField times;
+    private final ImList<JFormattedTextField> fields;
+
+    private final JRadioButton durationFixed;
+    private final JRadioButton durationForever;
+
+    private final JCheckBox repeatEnabled;
+
+    private final JRadioButton repeatTypeForever;
+    private final JRadioButton repeatTypeFixed;
+
+    private final ImList<JLabel> repeatLabels;
+
+    private final JButton okButton;
+    private final JButton cancelButton;
+
+    // Indicates if this dialog was cancelled.
+    private boolean cancelled;
```

Comment:
Yes, that should be the case. It should only ever be written to when the object is constructed and in `ActionListener`s for the OK and Cancel buttons.